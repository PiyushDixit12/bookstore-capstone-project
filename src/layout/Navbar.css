.background-dark {
    background-color: #333;
}

.background-transparent {
    background-color: transparent;
}

.navbar-container {
    position: absolute;
    width: 100%;
    top: 0;
    z-index: 1;
    padding: 1rem 0;
}

.logo {
    font-size: 1.5rem;
    color: var(--white-color);
    text-decoration: none;
    font-family: Chivo Mono;
    font-weight: 700;
}

.logo span {
    font-family: Chivo Mono;
}

.nav-links-container {
    display: flex;
    align-items: center;
}

.nav-links,
.nav-links-dark {
    color: var(--white-color);
    text-decoration: none;
    font-size: 1.2rem;
    padding-left: 20px;
    transition: 0.3s ease;
}

.nav-links-dark {
    color: #333;
}

.nav-links.active,
.nav-links-dark.active {
    color: var(--primary-color-bright);
}

.nav-links:hover,
.nav-links-dark:hover {
    color: var(--primary-color);
}
.close {
    display: none;
}

.mobile {
    display: none;
}

.menu-dark {
    color: #000;
    ;
}

.menu-light {
    color: white;
}


/*=========================== responsive navbar============================ */

@media screen and (max-width:460px) {
    .nav-links {
        font-size: 2rem;
    }
}

@media screen and (max-width: 768px) {
    .nav-links-container {
        display: flex;
        flex-direction: column;
        align-items: flex-start;
        justify-content: flex-start;
        position: fixed;
        z-index: 2;
        /* top: 0;
        right: -300px; */
        top: 0px;
        right: -100%;
        height: 100vh;
        width: 250px;
        background-color: #e3e6f3;
        box-shadow: 0 40px 60px rgba(0, 0, 0, 0.01);
        padding: 80px 0 0 10px;
        transition: 0.9s ease;
        align-self: flex-start;
    }
    .navbar-container {
        background: #000;
    }
    .nav-links-container.active {
        right: 0%;
        /* background: #000; */
        cursor: pointer;
        transition: 0.9s ease;
    }
    .nav-links,
    .nav-links-dark {
        color: black;
        width: 100%;
        /* margin-bottom: 25px; */
        margin-top: 25px;
        transition: 0.9s ease;
    }
    .mobile {
        display: flex;
        align-items: center;
        color: white;
    }
    .close {
        position: absolute;
        top: 30px;
        display: block;
        border: none;
        background-color: transparent;
        right: 20px;
        color: #222;
        font-size: 24px;
    }
}