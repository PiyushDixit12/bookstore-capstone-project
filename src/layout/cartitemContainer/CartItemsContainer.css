.cart-items-container {
    padding: 5% 0;
    min-height: 60vh;
}