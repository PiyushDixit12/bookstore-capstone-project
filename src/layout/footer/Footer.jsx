import React,{useRef} from 'react'
import './Footer.css'
import emailjs from '@emailjs/browser'
import {Field,Form,Formik} from 'formik';
import * as yup from 'yup';
import {toast} from 'react-toastify';
const serviceId = "service_hudclxo";
const templetId = "template_sg212jv";
const publicKey = "hmcfOZhjGCeHLqW1m";

const validationQuery = yup.object().shape({
    user_name: yup.string().min(2).max(100).required(),
    user_email: yup.string().email().required(),
    message: yup.string().min(5).max(1000).required()
});
export const Footer = () => {
    const formRef = useRef(null);
    console.log(formRef);
    async function handleEmailSubmit(e) {
    // e.preventDefault();
        console.log("email is sending ");
        await emailjs.sendForm(serviceId,templetId,formRef.current,publicKey).then((resp) => {
            console.log(resp);
            console.log("===================sended succeffuly ====================");
        }).catch(err => console.log(err));

    }

    return (
        <section data-testid="footer" className='footer-container'>
            <div className="container">
                <h2>If you have any query feel free to ask here</h2>
                <Formik
                    initialValues={{
                        user_name: "",
                        user_email: "",
                        message: ""
                    }}
                    validationSchema={validationQuery}
                    onSubmit={(values,{resetForm}) => {
                        handleEmailSubmit(values);
                        console.log(values);
                        toast.success("ðŸŽ‰ðŸŽ‰ Query submited successfully ðŸŽ‰ðŸŽ‰",{
                            position: "top-center",
                            autoClose: 5000,
                            hideProgressBar: false,
                            closeOnClick: true,
                            pauseOnHover: true,
                            draggable: true,
                            progress: undefined,
                            theme: "light",
                        });
                        resetForm();
                    }}
                >
                    {({handleBlur,handleChange,handleSubmit}) => {
                        return <Form ref={formRef} onSubmit={handleSubmit} className='footer-form'>
                    <div className="form-group">
                        <label htmlFor="name" className='form-label'>Name :</label>
                                <Field
                                    type="text"
                                    name='user_name'
                                    id='name'
                                    onBlur={handleBlur}
                                    onChange={handleChange}
                                    placeholder="Enter your name"
                                    className='form-input'
                                />
                    </div>
                            <div className="form-group">
                        <label htmlFor="email" className='form-label'>Email :</label>
                                <Field type="email"
                            id='email'
                            name='user_email'
                                    onBlur={handleBlur}
                                    onChange={handleChange}
                            placeholder="Enter your email"
                            className='form-input' />
                            </div>
                            <div className="form-group">
                        <label htmlFor="query" className='form-label'>Query : </label>
                                <Field className='form-input' name='message'
                                    onBlur={handleBlur}
                                    onChange={handleChange}
                                    placeholder="Type your Query" ></Field>
                    </div>
                    <div className="form-group">
                                <input type='submit' value="Submit" 
                                    className="form-submit" onSubmit={handleSubmit} />
                    </div>
                        </Form>
                    }
                    }
                </Formik>
                <p>&copy; 2023 bookstore. All Rights Reserved by Piyush Dixit</p>
            </div>
        </section>
    )
}
