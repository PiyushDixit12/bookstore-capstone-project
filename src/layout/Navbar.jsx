import React,{useContext,useState} from 'react'
import './Navbar.css'
import {Link,NavLink,useNavigate} from 'react-router-dom'
import {createUserWithEmailAndPassword,signInWithEmailAndPassword} from 'firebase/auth';
import {path} from '../routes/RouteConstant'
import {userContext} from '../App'
import {auth} from '../firebase/firebase'
import {signOut} from 'firebase/auth'
import {MdOutlineMenuOpen} from "react-icons/md";
import {GiSplitCross} from "react-icons/gi";
import {FaShoppingCart} from "react-icons/fa";

export const Navbar = ({darkTheme,darkTextTheme}) => {
    const [isNavbarOpen,setNavbarOpen] = useState(false);
    const toggleNavbar = () => {
        setNavbarOpen(!isNavbarOpen);
    };

    const user = useContext(userContext);
    console.log(user);
    const navigate = useNavigate();
    const logout = () => {
        console.log("logout called")
        signOut(auth).then((val) => {
            console.log("signout successfully")
            navigate(path.Home.path);
        }).catch((err) => {
            console.log("error",err);
        });
    }
    return (

        <div data-testid={"navbarContainer"} className={`navbar-container ${darkTheme ? 'background-dark position-relative' : 'background-transparent'} `}>
            <div className='container flex justify-between align-center'>

                <a href="/" className='logo'> Book<span className='text-primary'>store</span> </a>
                <div className={`mobile ${darkTextTheme ? 'menu-light-dark' : "menu-light"} `} onClick={toggleNavbar}>
                    <MdOutlineMenuOpen size={30} />
                </div>
                {user?.accessToken ? <nav className={`nav-links-container ${isNavbarOpen ? 'active' : ""}`}>
                    <button className='close' onClick={toggleNavbar}>  <GiSplitCross size={20} /></button>
                    <NavLink to={path.Home.path} className={` ${darkTextTheme ? 'nav-links-dark' : "nav-links"}`}> Home</NavLink>
                    <NavLink to={path.BooksPage.path} className={` ${darkTextTheme ? 'nav-links-dark' : "nav-links"}`}>Books</NavLink>
                    <Link className={` ${darkTextTheme ? 'nav-links-dark' : "nav-links"}`} onClick={logout} data-testid={"logout"}> logout</Link>
                    <NavLink to={path.cart.path} className={` ${darkTextTheme ? 'nav-links-dark' : "nav-links"}`}> <FaShoppingCart size={20} style={{marginTop: "5px"}} /></NavLink> 


                </nav> : <nav className={`nav-links-container ${isNavbarOpen ? 'active' : ""}`}>
                    <button className='close' onClick={toggleNavbar}> <GiSplitCross size={30} /></button>
                    <NavLink to={path.Home.path} className={` ${darkTextTheme ? 'nav-links-dark' : "nav-links"}`}> Home</NavLink>
                    <NavLink to={path.BooksPage.path} className={` ${darkTextTheme ? 'nav-links-dark' : "nav-links"}`}>Books</NavLink>
                    <NavLink to={path.Login.path} className={` ${darkTextTheme ? 'nav-links-dark' : "nav-links"}`}> Login</NavLink>      <NavLink to={path.SignUp.path} className={` ${darkTextTheme ? 'nav-links-dark' : "nav-links"}`}> sign up</NavLink>

                </nav>}


            </div>
        </div>
    )
}
