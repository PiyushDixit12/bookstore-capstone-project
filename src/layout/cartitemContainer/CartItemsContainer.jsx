import React,{useContext} from 'react'
import './CartItemsContainer.css'

import {CartItemCard} from '../../component/cards/cartItemsCard/CartItemCard'
import StripeCheckout from 'react-stripe-checkout'
import {useNavigate} from 'react-router-dom'
import {path} from '../../routes/RouteConstant'
import {useSelector} from 'react-redux'
const stripeKey = 'pk_test_51OONTjSFUXCQiIloeCA2pircU0F1E6N2wzwEahk9c3RkePIqY0M0qTMzrsczsSYyjeK4X3TNlGoOqjGrD4o6vC4800fsvzzEoO';
export const CartItemsContainer = () => {
    const navigate = useNavigate();
    const {cartItem,totalAmount} = useSelector(state => {
        console.log("cartitems containr",state.cart);
        return state.cart;
    })
    console.log(cartItem);
    const onToken = (token) => {
        console.log("token",token);
        alert("your Payment has been proccessed");
        navigate(path.BooksPage.path);
    }
    return (
        <section className='cart-items-container'>
            <div className="container">
                {totalAmount === 0 ? (
                    <h2>Currently cart is empty</h2>
                ) : (<>
                    <h2> Cart</h2>

                    {cartItem && cartItem?.map((book) => {
                        return <CartItemCard key={book?.id} bookData={book} />
                    })}
                    <h2>Total Amount = &#8377;{totalAmount}</h2>


                    <StripeCheckout
                        name='Book Checkout'
                        description='Please fill in the details below'
                        currency='INR'
                        amount={totalAmount * 100}
                        billingAddress
                        stripeKey={stripeKey}
                        token={onToken}
                    >   <button className='button-primary'>Procced to Checkout</button></StripeCheckout>
                </>)
                }
            </div >

        </section >
    )
}
