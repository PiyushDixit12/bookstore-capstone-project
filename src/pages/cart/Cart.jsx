import React from 'react'
import './Cart.css'
import {Navbar} from '../../layout/Navbar'
import {Footer} from '../../layout/footer/Footer'
import {CartItemsContainer} from '../../layout/cartitemContainer/CartItemsContainer'
export const Cart = () => {
    return (
        <section>
            <Navbar darkTheme={true} />
            <CartItemsContainer />
            <Footer />
        </section>
    )
}
