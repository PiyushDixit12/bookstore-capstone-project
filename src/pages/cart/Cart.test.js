
import React from 'react';
import {render,screen} from '@testing-library/react';
// import '@te/sting-library/jest-dom/extend-expect';
import {Cart} from './Cart';
import {Navbar} from '../../layout/Navbar';
import {Footer} from '../../layout/footer/Footer';
import {CartItemsContainer} from '../../layout/cartitemContainer/CartItemsContainer';

jest.mock('../../layout/Navbar',() => ({
    Navbar: jest.fn(() => null)
}));

jest.mock('../../layout/footer/Footer',() => ({
    Footer: jest.fn(() => null)
}));

jest.mock('../../layout/cartitemContainer/CartItemsContainer',() => ({
    CartItemsContainer: jest.fn(() => null)
}));

describe('Cart Page',() => {
    it('should render without crashing',() => {
        render(<Cart />);
    });

    it('should render the Navbar component with correct props',() => {
        Navbar.mockImplementation(() => <div data-testid="navbar"></div>);

        render(<Cart />);

        expect(screen.getByTestId('navbar')).toBeInTheDocument();
        expect(Navbar).toHaveBeenCalledWith({darkTheme: true},{});
    });

    it('should render the CartItemsContainer component',() => {
        CartItemsContainer.mockImplementation(() => <div data-testid="cartitemscontainer"></div>);

        render(<Cart />);

        expect(screen.getByTestId('cartitemscontainer')).toBeInTheDocument();
        expect(CartItemsContainer).toHaveBeenCalled();
    });

    it('should render the Footer component',() => {
        Footer.mockImplementation(() => <div data-testid="footer"></div>);

        render(<Cart />);

        expect(screen.getByTestId('footer')).toBeInTheDocument();
        expect(Footer).toHaveBeenCalled();
    });

    it('should match snapshot',() => {
        Navbar.mockImplementation(() => <div>Navbar</div>);
        CartItemsContainer.mockImplementation(() => <div>CartItemsContainer</div>);
        Footer.mockImplementation(() => <div>Footer</div>);

        const {asFragment} = render(<Cart />);

        expect(asFragment()).toMatchSnapshot();
    });
});
