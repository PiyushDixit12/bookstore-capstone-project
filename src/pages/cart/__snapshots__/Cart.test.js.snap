// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cart Page should match snapshot 1`] = `
<DocumentFragment>
  <section>
    <div>
      Navbar
    </div>
    <div>
      CartItemsContainer
    </div>
    <div>
      Footer
    </div>
  </section>
</DocumentFragment>
`;

exports[`Cart should match snapshot 1`] = `
<DocumentFragment>
  <section>
    <div>
      Navbar
    </div>
    <div>
      CartItemsContainer
    </div>
    <div>
      Footer
    </div>
  </section>
</DocumentFragment>
`;
